example/lib/widgets/location_page.dart

```diff
@@ -0,0 +1,208 @@
+                        fontSize: Styles.locationCardTitleFontSize,
+                        fontWeight: FontWeight.w500,
+                        color: Colors.grey.shade100,
+                        shadows: [
+                          Shadow(
+                            color: Colors.black,
+                            blurRadius: Styles.locationCardTextShadowBlur,
+                          ),
+                        ],
+                      ),
+                      textAlign: TextAlign.left,
+                    ),
+                  ],
+                ),
+                StreamBuilder(
+                  stream: AnotherLocationPlugin.locationStream,
+                  initialData: Strings.locationCardHint,
+                  builder: (context, snapshot) {
+                    return snapshot.hasData
+                        ? Row(
+                            mainAxisAlignment: MainAxisAlignment.start,
+                            children: [
+                              Container(
+                                width: Dimensions.locationCardHintContainerSize,
+                                child: Text(
+                                  '${snapshot.data}',
+                                  style: TextStyle(
+                                    fontSize: Styles.locationCardHintFontSize,
+                                    fontWeight: FontWeight.w700,
+                                    color: Colors.grey.shade100,
+                                    shadows: [
+                                      Shadow(
+                                        color: Colors.black,
+                                        blurRadius:
+                                            Styles.locationCardTextShadowBlur,
+                                      ),
+                                    ],
+                                  ),
+                                ),
+                              ),
+                            ],
+                          )
+                        : Text(
+                            Strings.locationEmptyText,
+                          );
+                  },
+                ),
+                Row(
+                  children: [
+                    Column(
+                      crossAxisAlignment: CrossAxisAlignment.start,
+                      children: [
+                        MainButton(
+                          onPressedFunction: isInitiallized
+                              ? () {
+                                  AnotherLocationPlugin.lastCoordinates;
+                                }
+                              : null,
+                          children: [
+                            Icon(
+                              Icons.location_on,
+                            ),
+                            Text(
+                              Strings.locationCardFindButtonText,
+                            ),
+                          ],
+                        ),
```

Comment:
This logic is not necessary you can make it all inside MainButton. Instead to do this you can pass as parameter 4 things:
isInitialized, callBackFunction, Icons.location_on, Strings.locationCardFindButtonText

Reply:
Refactored: now all the logic is inside the MainButton and receives parameters